### ISSUE TITLE
router with strict_slashes disabled doesn't match URL without slash

### ISSUE THREAD

**Issue by pjulian** [pallets/werkzeug#2482]
Under v2.1.0 and earlier, `strict_slashes=False` could be used to make a routing rule for `/foo/` match `/foo` as well.
This appears to have changed in 2.2.0. (**NOTE**: opposite of what was reported in bug report -> https://github.com/pallets/werkzeug/issues/2467)

Example flask app:

```
# app.py
import flask
app = flask.Flask(__name__)
app.url_map.strict_slashes = False

@app.route("/foo/")
def foo():
    return "hi"
```

... And then flask run and curl localhost:5000/foo. Under v2.1.0 you should see hi, but under 2.2.0, a 404.

My app currently sets strict_slashes = False and all routes are returning a 404.

Environment:

- Python version: 3.9
- Werkzeug version: 2.2.x

**Comment by pgjones** [comment]
Duplicate of #2478?

**Comment by pjulian** [comment]
I *might* possibly agree that #2478 *could* be a bug report of the same issue ... 

Correct me if I'm wrong, but the resolution of #2478 appears to be that the behavior WAS wrong before and has been FIXED in #1074?  And all that was needed was to just update the change log to reflect the change in behavior?

The reporter in #2478 seems to be okay with simply updating the change log, but I'm not sure if I agree.

IMO, EITHER of the following should resolve to the same route (currently the latter is treated as if `strict_slashes` were `True` and causing `/foo` to return a 404):

Reported in #2467:
```
# app.py
import flask
app = flask.Flask(__name__)
app.url_map.strict_slashes = False

@app.route("/foo")
def foo():
    return "hi"
```

Being reported here #2482 (and I guess also possibly here #2478):
```
# app.py
import flask
app = flask.Flask(__name__)
app.url_map.strict_slashes = False

@app.route("/foo/")
def foo():
    return "hi"
```

**Comment by pgjones** [comment]
I see. I think #2487 is the solution to these issues. There are really three options people expect.
